frontend/src/pages/notebookController/screens/server/EnvironmentVariablesField.tsx

```diff
@@ -42,40 +46,45 @@ const EnvironmentVariablesField: React.FC<EnvironmentVariablesFieldProps> = ({
   const validated = variableRow.errors[variable.name] !== undefined ? 'error' : 'default';
   return (
     <div className="odh-notebook-controller__env-var-field">
-      <FormGroup
-        fieldId={`${fieldIndex}-${variable.name}`}
-        label="Variable name"
-        helperTextInvalid={variableRow.errors[variable.name]}
-        helperTextInvalidIcon={<ExclamationCircleIcon />}
-        validated={validated}
-      >
+      <FormGroup fieldId={`${fieldIndex}-${variable.name}`} label="Variable name">
         <TextInput
           id={`${fieldIndex}-${variable.name}`}
           data-id={`${fieldIndex}-${variable.name}`}
           type={TextInputTypes.text}
-          onChange={(newKey) =>
+          onChange={(e, newKey) =>
             onUpdateVariable({ name: newKey, type: variable.type, value: variable.value })
           }
           value={variable.name === EMPTY_KEY ? '' : variable.name}
           validated={validated}
         />
+        {validated === 'error' && (
+          <FormHelperText>
+            <HelperText>
+              <HelperTextItem icon={<ExclamationCircleIcon />} variant={'error'}>
+                {variableRow.errors[variable.name]}
+              </HelperTextItem>
+            </HelperText>
+          </FormHelperText>
+        )}
       </FormGroup>
       <FormGroup fieldId={`${fieldIndex}-${variable.name}-value`} label="Variable value">
         <Flex>
           <InputGroup>
-            <TextInput
-              id={`${fieldIndex}-${variable.name}-value`}
-              data-id={`${fieldIndex}-${variable.name}-value`}
-              type={
-                showPassword && variableType === 'password'
-                  ? TextInputTypes.text
-                  : (variable.type as TextInputTypes)
-              }
-              value={variable.value}
-              onChange={(newValue) =>
-                onUpdateVariable({ name: variable.name, type: variable.type, value: newValue })
-              }
-            />
+            <InputGroupItem isFill>
+              <TextInput
+                id={`${fieldIndex}-${variable.name}-value`}
+                data-id={`${fieldIndex}-${variable.name}-value`}
+                type={
+                  showPassword && variableType === 'password'
+                    ? TextInputTypes.text
+                    : (variable.type as TextInputTypes)
+                }
+                value={variable.value}
+                onChange={(e, newValue) =>
+                  onUpdateVariable({ name: variable.name, type: variable.type, value: newValue })
+                }
+              />
+            </InputGroupItem>
             {variable.type === 'password' ? (
```

Comment:
Before:
<img width="814" alt="Screenshot 2023-10-10 at 3 03 21 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/8126518/135cff1c-ef63-4e6d-bbef-a70478a5eb6d">

After:
![Screenshot 2023-10-10 at 3 03 30 PM](https://github.com/opendatahub-io/odh-dashboard/assets/8126518/353f7e52-4f11-4b15-9a06-8f1ca1640e63)

Reply:
I am unable to reproduce both the topology issue and this environment variables issue in my local instance of the shared cluster. Would be great if the team could help investigate these issues cc @andrewballantyne